main.tf

```diff
@@ -121,6 +126,90 @@ resource "aws_cloudwatch_log_group" "main" {
   }
 }
 
+resource "aws_cloudwatch_metric_alarm" "alarm_cpu" {
+  count = "${var.cloudwatch_alarm_cpu_enable && (var.associate_alb || var.associate_nlb) ? 1 : 0}"
+
+  alarm_name        = "${local.cloudwatch_alarm_name}-cpu"
+  alarm_description = "Monitors ECS CPU Utilization"
+  alarm_actions     = ["${var.cloudwatch_alarm_actions}"]
+
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = "2"
+  metric_name         = "CPUUtilization"
+  namespace           = "AWS/ECS"
+  period              = "120"
+  statistic           = "Average"
+  threshold           = "${var.cloudwatch_alarm_cpu_threshold}"
+
+  dimensions = {
+    "ClusterName" = "${var.ecs_cluster_name}"
+    "ServiceName" = "${aws_ecs_service.main.name}"
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "alarm_mem" {
+  count = "${var.cloudwatch_alarm_cpu_enable && (var.associate_alb || var.associate_nlb) ? 1 : 0}"
+
+  alarm_name        = "${local.cloudwatch_alarm_name}-mem"
+  alarm_description = "Monitors ECS CPU Utilization"
+  alarm_actions     = ["${var.cloudwatch_alarm_actions}"]
+
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = "2"
+  metric_name         = "MemoryUtilization"
+  namespace           = "AWS/ECS"
+  period              = "120"
+  statistic           = "Average"
+  threshold           = "${var.cloudwatch_alarm_mem_threshold}"
+
+  dimensions = {
+    "ClusterName" = "${var.ecs_cluster_name}"
+    "ServiceName" = "${aws_ecs_service.main.name}"
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "alarm_cpu_no_lb" {
```

Comment:
Can you explain why we need duplicate alarms depending on whether we do or don't have an LB?

Reply:
Yes I can! I originally had a ternary for the `ServiceName` in the dimension. But you can't say "use this resource or that resource" if one of them is never actually created due to the `count` value being 0. So instead I had to make separate metric alarms. Only two of the four will ever be created and its entirely based on the use of the LB or not.

Reply:
Got it. I have run into stuff like that before with Terraform, I guess that's not super surprising. :)